smoketest/directives.py

```diff
@@ -267,11 +271,22 @@ def generate_directives(self):
 
     def _generate_directives_from_json_or_yaml(self):
         # Load input
-        with open(self.filename, 'r') as file_:
-            if self.filename.endswith('json'):
-                input_ = json.load(file_)
-            else:
-                input_ = yaml.safe_load(file_)
+        try:
+            with open(self.filename, 'r') as file_:
+                if self.filename.endswith('json'):
+                    try:
+                        input_ = json.load(file_)
+                    except ValueError as e:
+                        # This happens if the JSON was invalid.
```

Comment:
Nitpick: I think it would be nice if we picked between comments ending in a period and comments not ending in a period. The current mix of inconsistent comment punctuation, especially in the same file, bothers me a little.

Reply:
Okay, it looks like most of the comments in this file have periods, so I added went with that in 26ed296.